R/draws.R

```diff
@@ -1,10 +1,112 @@
-#' TODO
+#' @title Fit the base imputation model and get parameter estimates
+#'
+#' @description Assuming MAR, `draws` fits the base imputation model to the observed data
+#' according to the given multiple imputation methodology and returns the parameter
+#' estimates from the fit.
 #'
 #' @name draws
-#' @param data TODO
-#' @param data_ice TODO
-#' @param vars TODO
-#' @param method TODO
+#' @param data A `data.frame` containing the data to be used in the model.
+#' Must contain the outcome variable. See details.
+#' @param data_ice A `data.frame` that specifies the information related
+#' to the intercurrent events (ICE) and the imputation strategies. Its structure is described in the details.
+#' @param vars A `vars` object as generated by [set_vars()]. See details.
+#' @param method An object returned by the functions [method_bayes()],
+#' [method_approxbayes()] or [method_condmean()].
+#' It specifies the multiple imputation methodology to be used. See details.
+#'
+#' @details
+#'
+#' `draws` performs the first step of the multiple imputation (MI) procedure: the fit of the
+#' base imputation model. The goal is to estimate the parameters of interest needed
+#' for the imputation phase (i.e. the regression coefficients and the covariance matrices
+#' from a MMRM model).
+#'
+#' The function distinguishes between the following methods:
+#' - Bayesian MI based on MCMC sampling: `draws` returns the draws
+#' from the posterior distribution of the parameters using a Bayesian approach based on
+#' MCMC sampling. This method can be adopted by specifying `method` using [method_bayes()].
+#' - Approximate Bayesian MI based on bootstrap: `draws` returns
+#' the draws from the posterior distribution of the parameters using a Bayesian approach,
+#' where the sampling from the posterior distribution is simulated fitting the MMRM model
+#' on bootstrap samples of the original dataset. This method can be adopted by
+#' specifying `method` through [method_approxbayes()].
+#' - Conditional mean imputation with bootstrap re-sampling: `draws` returns the
+#' ML/REML parameter estimates from the original dataset and from `n_samples` bootstrap samples.
+#' This method can be adopted by specifying `method` through [method_condmean()] with
```

Comment:
(pedantic comment) not that keen on the word adopted. Perhaps

> This method can be specified by using `method = method_condmean()`

Reply:
done